annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -213,6 +213,24 @@ public AnnotatedRecord convertResponseToAnnotatedRecord(VariantAnnotation gnResp
         // get the canonical transcript
         TranscriptConsequenceSummary canonicalTranscript = getCanonicalTranscript(gnResponse);
 
+        String resolvedReferenceAllele = annotationUtil.resolveReferenceAllele(gnResponse, mRecord);
+        String resolvedTumorSeqAllele1 = mRecord.getTUMOR_SEQ_ALLELE1();
+        String resolvedTumorSeqAllele2 = annotationUtil.resolveTumorSeqAllele(gnResponse, mRecord);
+
+        // Copy over changes to the reference allele or tumor_seq_allele1 if
+        // they were identical in the input
+        if (mRecord.getTUMOR_SEQ_ALLELE1().equals(mRecord.getREFERENCE_ALLELE())) {
+            resolvedTumorSeqAllele1 = resolvedReferenceAllele;
+        } else if (mRecord.getTUMOR_SEQ_ALLELE1().equals(mRecord.getTUMOR_SEQ_ALLELE2())) {
+            resolvedTumorSeqAllele1 = resolvedTumorSeqAllele2;
+        } else {
+            // TODO: it's also possible that the position has changed after
+            // resolving ref+alt. Tumor seq allele1 would have to be updated
+            // then as well. Kind of a corner case, but we might want to handle
+            // that in some way. Discussion point: should we even allow the core
+            // variant attributes (pos,ref,alt1,alt2) to be mutable?
```

Comment:
This is the question of the last few years. It's the same behavior that vcf2maf/maf2maf supported right? If we don't alter the position(s) then the annotation for the variant would fail based on VEP's reference genome sequence for the input variant's coordinates not matching the reference allele that we are submitting for annotation. 

So do we continue our current approach and allow the core variant attributes to be mutable or do we fail those annotations and require the user to fix and/or adjust them themselves?